### DISCUSSION TITLE
Will `zx` start using the new `--experimental-detect-module`?

### DISCUSSION THREAD

**Discussion by josh-hemphill** [google/zx#698] [General]
Being able to write extensionless esm scripts by default would be wonderful. Seems like `--experimental-detect-module` would let that be a reality.

**Comment by antongolub** [comment]
Why is it necessary to trim extensions in your case?

**Comment by antongolub** [comment]
Seems that extensionless files are properly processed now.

https://github.com/google/zx/blob/cb6f205d24bf4dd4bb17bee8386f1c7faf5e7bd8/src/cli.ts#L203-L206
